frontend/public/components/pod-logs.jsx

```diff
@@ -3,16 +3,37 @@ import * as React from 'react';
 
 import { Dropdown, ResourceLog, ResourceName } from './utils';
 
-// Component to container dropdown or conatiner name if only one container in pod.
-const ContainerDropdown = ({currentContainer, containers, kind, onChange}) => {
-  const resourceName = (container) => {
-    return <ResourceName name={container.name} kind={kind} />;
+const reduceContainerStatuses = (accumulator, container, order) => {
+  return {
+    ...accumulator,
+    [container.name]: { ...container, order }
   };
-  const dropdownItems = _.mapValues(containers, resourceName);
+};
+
+const containerLabel = (container) => {
+  return <ResourceName name={container.name} kind="Container" />;
+};
+
+const ContainerDropdown = ({currentKey, containers, initContainers = {}, onChange}) => {
+  let spacerBefore = new Set();
```

Comment:
Why create `new Set()` when you are only adding a single item?

Reply:
A value in the Set may only occur once; it is unique in the Set's collection.

Seems to be the reason to use a set per [MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set).

Reply:
This is because the `spacerBefore` prop of the `Dropdown` component is a Set.

Reply:
Well, you could possibly add multiple separators to a dropdown, so it makes sense that the Set ensures only one instance of each key exists. It just happens that we are only using one separator in this case.

Reply:
Got it, that makes sense then.